### ISSUE TITLE
app: Pistol bullet damage effect

### ISSUE THREAD

**Issue by amnaarshed4224** [webaverse/app#2034]
Currently, the Damage caused by pistol pullet appears hardcode at objects position. If the object is moveable such as a collectable, the damage effect seems suspended in the air while the object looks just fine.
shown in the two videos below

https://user-images.githubusercontent.com/35926530/146655809-577a55ef-dc40-4d4e-97e4-83d710cc19cc.mp4

https://user-images.githubusercontent.com/35926530/146655813-1be3dc69-d710-4e04-af2d-57be2f8bea31.mp4

.

**Comment by avaer** [comment]
Yes, we should attach the decal meshes to the parent object with e.g. a `WeakMap`.
